File: KampIntro/30-InterfaceAbstractDemo/Concrete/StarbucksCustomerManager.cs
using _30_InterfaceAbstractDemo.Abstract;
using _30_InterfaceAbstractDemo.Entities;
using System;
using System.Collections.Generic;
using System.Text;
using System.Linq;
using System.Threading.Tasks;

namespace _30_InterfaceAbstractDemo.Concrete {
    public class StarbucksCustomerManeger : BaseCustomerManager {
        private ICustomerCheckService _customerCheckService;

        public StarbucksCustomerManeger(ICustomerCheckService customerCheckService) {
            _customerCheckService = customerCheckService;
        }

        public override void Save(Customer customer) {
            if (_customerCheckService.CheckIfRealPerson(customer)) {
                base.Save(customer);
            }
            else {
                Console.WriteLine("Not a valid person: " + customer.FirstName);
            }
        }
    }
}

File: Java - React/Java/33-Homework2/src/interfaceAbstractDemo/Concrete/StarbucksCustomerManager.java
package interfaceAbstractDemo.Concrete;

import interfaceAbstractDemo.Abstract.BaseCustomerManager;
import interfaceAbstractDemo.Abstract.CustomerCheckService;
import interfaceAbstractDemo.Entities.Customer;

public class StarbucksCustomerManager extends BaseCustomerManager{
	CustomerCheckService customerCheckService;

	public StarbucksCustomerManager(CustomerCheckService customerCheckService) {
		this.customerCheckService = customerCheckService;
	}

	@Override
	public void save(Customer customer) {
		if(customerCheckService.checkIfRealPerson(customer)) {
			System.out.println("Starbucks Manager : Validation successfull.");
			super.save(customer);
		}else {
			System.out.println("Starbucks Manager : Validation Error - Not a valid person.");
		}

	}
	
}
